### DISCUSSION TITLE
Odd magit-cherry-spinout behavior

### DISCUSSION THREAD

**Discussion by isadon** [magit/magit#4351] [Support] [answered]
Everytime I mark 1 cherry from branch A to `magit-cherry-spinout` to new branch B magit asks: 

`Create branch from 1 cherries starting at:`

Upon hitting `<Enter>` the default (at least in helm) is always branch A (HEAD with the cherries). Keep note of this..

Magit then fails the operation with the following message:

```
  0 git … branch B A
Branch 'B' set up to track local branch 'A'.

  0 git … checkout B
Switched to branch 'B'
Your branch is up to date with 'A'.

  1 git … cherry-pick --ff e9a4270
On branch B
Your branch is up to date with 'A'.

You are currently cherry-picking commit e9a4270.
  (all conflicts fixed: run "git cherry-pick --continue")
  (use "git cherry-pick --skip" to skip this patch)
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

nothing to commit, working tree clean
The previous cherry-pick is now empty, possibly due to conflict resolution.
If you wish to commit it anyway, use:

    git commit --allow-empty

Otherwise, please use 'git cherry-pick --skip'
```

Its obvious that the operation fails because `magit-cherry-spinout` is checking out branch B by pointing to A at its HEAD which includes the cherries but when it comes time cherry pick there are no cherries to pick. The cherries are already in branch B. 

My question is: why does `magit-cherry-spinout` not assume here: 

`Create branch from 1 cherries starting at:`

that the starting point is branch `A~1` by default? When picking more than x cherries I have to remember to say `A~x` everytime and possibly mess that up if you aren't paying attention.. Spinout to me implies that a move will happen but how does the default being branch `A` and then causing the failed cherry pick above go in line with that?

**Comment by tarsius** [comment, marked as answer]
You probably want to use `magit-branch-spinoff` (`b S`) instead.
